sBinNode/sBinNode.c

```diff
@@ -0,0 +1,55 @@
+#include "sBinNode.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+
+
+sBinNode* CreateBinNode(const char* str)
+{
+	sBinNode* father = (sBinNode*) malloc(sizeof(sBinNode));
```

Comment:
This `malloc` can fail, it's good practice to check return values.

Reply:
Also, it might be just me, but forcing the user to use libc's general purpose allocator (`malloc` and `free`) isn't recommended since a lot of projects like to "roll their own allocator". Consider switching from "allocating" functions to "initializing" functions, and allow the user to choose his own allocator.

Notice that this will change the implementation of `FreeMemory` as well!